crates/core/src/factories/core_fetch.rs

```diff
@@ -0,0 +1,341 @@
+                            inner.purge_cool_down_list(Instant::now()).await;
+
+                            // Do not send request to agent on cool-down list.
+                            if !inner
+                                .cool_down_list
+                                .lock()
+                                .await
+                                .contains_key(&agent_id)
+                            {
+                                // Register new request, increase request count.
+                                (*current_request_count.lock().await) += 1;
+
+                                // Make new request in a separate task.
+                                tokio::spawn({
+                                    let inner = inner.clone();
+                                    let op_id = op_id.clone();
+                                    let agent_id = agent_id.clone();
+                                    let transport = transport.clone();
+                                    let current_request_count =
+                                        current_request_count.clone();
+                                    async move {
+                                        let result = transport
+                                            .lock()
+                                            .await
+                                            .send_op_request(
+                                                op_id.clone(),
+                                                agent_id.clone(),
+                                            )
+                                            .await;
+
+                                        // Request is complete, decrease request count.
+                                        (*current_request_count
+                                            .lock()
+                                            .await) -= 1;
+
+                                        if let Err(err) = result {
+                                            eprintln!("could not send fetch request for op id {op_id} to agent {agent_id}: {err}");
+                                            inner
+                                                .cool_down_list
+                                                .lock()
+                                                .await
+                                                .insert(
+                                                    agent_id,
+                                                    Instant::now(),
+                                                );
+                                        }
+                                    }
+                                });
+                            }
+
+                            // Pause to give other requests a chance to be processed before re-adding this op.
+                            tokio::time::sleep(Duration::from_millis(1)).await;
```

Comment:
Is this a meaningful amount of delay? It seems quite short.

Reply:
There's no longer a delay.